rure/lib.py

```diff
@@ -250,3 +250,88 @@ def shortest_match(self, haystack, start=0):
         end = ffi.new('size_t *')
         if _lib.rure_shortest_match(self._ptr, haystack, hlen, start, end):
             return end[0]
+
+
+class RureSet(object):
+    """ Match multiple (possibly overlapping) regular expressions in a single
+    scan.
+
+    A regex set corresponds to the union of two or more regular expressions.
+    That is, a regex set will match text where at least one of its constituent
+    regular expressions matches. A regex set as its formulated here provides a
+    touch more power: it will also report which regular expressions in the set
+    match. Indeed, this is the key difference between regex sets and a single
+    Regex with many alternates, since only one alternate can match at a time.
+    """
+    def __init__(self, res, flags=DEFAULT_FLAGS, **options):
+
+        """ Compiles a regular expression. Once compiled, it can be used
+        repeatedly to search, split or replace text in a string.
+
+        :param res:     List of Bytestring expressions to compile
+        :param flags:   Bitmask of flags
+        :param kwargs:  Config options to pass (size_limit, dfa_size_limit)
+        """
+
+        if not all(isinstance(re, bytes) for re in res):
```

Comment:
This should probably check to ensure `res` is not a string object before iterating. For instance, on Python 2.7:
```python
>>> all([isinstance(i, bytes) for i in b'foobar'])
True
```

Python 3 behaves differently here, as iterating over a byte string returns the integer representing the Unicode code point of that character:

```python
>>> all([isinstance(i, bytes) for i in b'foobar'])
False
```